lib/src/themes/theme_layers.dart

```diff
@@ -21,11 +24,41 @@ class DefaultLayer extends ThemeLayer {
 
   @override
   void render(Context context) {
-    for (final feature
-        in context.tileset.resolver.resolveFeatures(this.selector)) {
-      context.featureRenderer
-          .render(context, type, style, feature.layer, feature.feature);
+    final layers = selector.select(context.tileset).toList(growable: false);
+    assert(layers.length <= 1);
+
+    if (layers.isEmpty) {
+      return;
+    }
+
+    final features = context.tileset.resolver
+        .resolveFeatures(this.selector)
+        .toList(growable: false);
+
+    if (features.isEmpty) {
+      return;
     }
+
+    final layer = layers.first;
+
+    context.tileSpaceMapper = TileSpaceMapper(
```

Comment:
I wanted to avoid making `tileSpaceMapper` nullable because when it is actually used it is never `null`. We have to set it in the `ThemeLayer` because before that, we don't know what extent the layer has we are rendering. A tileset could have tiles with different extents.